Convert IKEA Styrbar quirks to v2 quirk

TheJulianJES opened the pull request:
## Proposed change
Converts existing IKEA Styrbar quirks to a v2 quirk.
This also adds support for new signatures of the Styrbar, as quirks v2 don't need to match by endpoint/clusters (signature) anymore.


## Additional information
The only downside I see is that we use `DoublingPowerConfig2AAACluster` for all versions of the remote, even if it's a "newer signature". I think we should at least change the default to not double for that first, before we merge this.
Related PR for that:
- https://github.com/zigpy/zha-device-handlers/pull/3176

Would supersede #3144 if we want to go this route.
The test `test_quirk_device_automation_triggers_unique` also still needs updating that should be done in a separate PR.
It doesn't deal properly with v2 quirks.

## Checklist

- [ ] The changes are tested and work correctly
- [x] `pre-commit` checks pass / the code has been formatted using Black
- [ ] Tests have been added to verify that the new code works

TheJulianJES committed:
Convert IKEA Styrbar quirks to v2 quirk

MattWestb commented:
OnOff and Level looks working OK then doing fast testing.
Scene is not working then you was deleting the "IKEA Scene" cluster replacement so need adding it back for getting some functions on left and right buttons.
Have not testing battery or deeper.

I like V2 quirks its much cleaner !!!

TheJulianJES commented:
Ah, yeah.
I've re-added the `ScenesCluster` now. Can you try the updated variant?

TheJulianJES committed:
Remove `fourbtnremote` from `test_quirk_device_automation_triggers_unique`

The test does not handle v2 quirks yet and we'll need to see how we want to implement this, as all v2 quirks share the same quirk class.
For now, this just removes the `fourbtnremote` from the test.

TheJulianJES committed:
Also replace `ScenesCluster`

This is needed for left/right buttons to work properly

MattWestb commented:
Was updating it from your git and now i getting "scene commands" OK.
But i think long press up and down is not working OK then we is not using all parameter (not in the device log or zha_events but have not testing the DA):
```
event_type: zha_event
data:
  device_ieee: 8c:f6:81:ff:fe:1e:f3:70
  unique_id: 8c:f6:81:ff:fe:1e:f3:70:1:0x0008
  device_id: 64a05efdf2bd3e0e6e80f89e7c7bf5b4
  endpoint_id: 1
  cluster_id: 8
  command: move
  args:
    - 1
    - 83
    - 0
    - 0
  params:
    move_mode: 1
    rate: 83
    options_mask: 0
    options_override: 0
origin: LOCAL
time_fired: "2024-05-29T15:30:34.903406+00:00"
context:
  id: 01HZ2FJY8QG68NGJZAKA3ZSSED
  parent_id: null
  user_id: null

 event_type: zha_event
data:
  device_ieee: 8c:f6:81:ff:fe:1e:f3:70
  unique_id: 8c:f6:81:ff:fe:1e:f3:70:1:0x0008
  device_id: 64a05efdf2bd3e0e6e80f89e7c7bf5b4
  endpoint_id: 1
  cluster_id: 8
  command: move_with_on_off
  args:
    - 0
    - 83
  params:
    move_mode: 0
    rate: 83
origin: LOCAL
time_fired: "2024-05-29T15:30:39.254780+00:00"
context:
  id: 01HZ2FK2GPKNGR175CY9APVBQ9
  parent_id: null
  user_id: null
```
Also some users dont like that we is not remember the long press for the release like Open/Close is doing here:

https://github.com/TheJulianJES/zha-device-handlers/blob/9ed616a33e6f973f57f1aec37d67cabcff754aaa/zhaquirks/ikea/opencloseremote.py#L47-L76

TheJulianJES commented:
Ok, thanks. But those aren't regressions, right? So no new bugs introduced with this?

TheJulianJES commented:
Codecov project coverage decrease is expected, as the old quirk had quite a few lines and is now fully removed with this.

MattWestb commented:
> Ok, thanks. But those aren't regressions, right? So no new bugs introduced with this?

I think it was not working OK then we was chaining par/args and it was very much testing for all IKEA controllers then doing that and my memory . . .

I dont have any problems then i using them as Light controller with device and group bindings so its working then ZHA is not OK.

MattWestb commented:
Thanks for fixing and closing my V3 that is out of date now !!!

MattWestb commented:
Only for information:

Dirigeras feed have now updated to 0x02040016 and the quirk is working great and then the new firmware have getting one new private cluster but sadly no group binding is being added like all newer IKEA gen 3 controller is having.

Edit: 2 of 4 inn production was loosing there binding to deices and must being redone  for working after the upgrade.
